### PR TITLE
[WebApp] Create earning line chart

### PR DESCRIPTION
**Pull request by Maks19**
**Description:** Create earning line chart

**Issue:** https://www.notion.so/saladtech/Create-new-line-chart-for-multiple-IDs-5219629c3455462fb62f081f7e844d29

image

### CODE REVIEW

File: packages/web-app/src/modules/earn-views/EarningLineChartContainer.tsx [hunk 1]
```diff
@@ -0,0 +1,1254 @@
+import { connect } from '../../connect'
+import type { RootStore } from '../../Store'
+import { EarningLineChart } from './components/EarningLineChart'
+
+const mockedEarningsPerMachine = (daysShowing: number) => {
+  const shouldShowAmPm = daysShowing === 1
+  const shouldShowDateMonth = daysShowing === 7
+  const earningsPerMachine24hours = {
```
Comment (vitto-moz) [inline packages/web-app/src/modules/earn-views/EarningLineChartContainer.tsx:8]: this type is not aligned with the one we have fetched with Kiota 
```

File: packages/web-app/src/modules/earn-views/components/EarningChart.tsx [hunk 2]
```diff
@@ -180,7 +180,7 @@ interface EarningRange {
 }
 
 interface CustomTick extends WithStyles<typeof styles> {
```
Comment (vitto-moz) [inline packages/web-app/src/modules/earn-views/components/EarningChart.tsx:182]:
```suggestion
interface CustomizedXAxisTickProps extends WithStyles<typeof styles> {
```

File: packages/web-app/src/modules/earn-views/components/EarningLineChart.tsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+interface CustomTick extends WithStyles<typeof styles> {
+  x: number
+  y: number
+  payload: {
+    value: string
+  }
+  fill: string
+  is24HoursChart: boolean
+}
```
Comment (vitto-moz) [inline packages/web-app/src/modules/earn-views/components/EarningLineChart.tsx:34]: looks like we don't pass something except of `is24HoursChart` in this component: do we still need all other properties? 

`<CustomizedXAxisTick is24HoursChart={is24HoursChart} />`

File: packages/web-app/src/modules/earn-views/components/EarningLineChart.tsx [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+const CustomizedXAxisTick = (props: CustomTick) => {
+  const { x, y, fill, payload, is24HoursChart } = props
+  if (!payload.value || payload.value === '0') {
+    return null
+  }
+
+  const shouldShowAmPm = payload.value === '00' || payload.value === '12'
```
Comment (vitto-moz) [inline packages/web-app/src/modules/earn-views/components/EarningLineChart.tsx:42]: Can we replace '00' and '12' with some meaningful constants?

**Review by vitto-moz — CHANGES_REQUESTED** [review]

File: packages/web-app/src/modules/earn-views/components/EarningLineChart.tsx (hunk 3 above)
Comment (Maks19) [reply packages/web-app/src/modules/earn-views/components/EarningLineChart.tsx:34]: It's passed from XAxis and I use these props inside CustomizedXAxisTick

File: packages/web-app/src/modules/earn-views/components/EarningLineChart.tsx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    </text>
+  )
+}
+
+interface Props extends WithStyles<typeof styles> {
+  earningsPerMachine: EarningPerMachine
+  daysShowing: ChartDaysShowing
+}
+
+const _EarningLineChart = ({ classes, earningsPerMachine, daysShowing }: Props) => {
```
[the remark this answers is not in the document]
Comment (Maks19) [reply packages/web-app/src/modules/earn-views/components/EarningLineChart.tsx:63]: I believe it's comment duplication 
Moved CustomizedXAxisTick  to it's own file

File: packages/web-app/src/modules/earn-views/components/EarningLineChart.tsx (hunk 4 above)
Comment (Maks19) [reply packages/web-app/src/modules/earn-views/components/EarningLineChart.tsx:42]: Done

File: packages/web-app/src/modules/earn-views/components/EarningLineChart.tsx [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+  x: number
+  y: number
+  payload: {
+    value: string
+  }
+  fill: string
+  is24HoursChart: boolean
+}
+
+const CustomizedXAxisTick = (props: CustomTick) => {
```
[the remark this answers is not in the document]
Comment (Maks19) [reply packages/web-app/src/modules/earn-views/components/EarningLineChart.tsx:36]: Done

File: packages/web-app/src/modules/earn-views/components/EarningLineChart.tsx [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+import type { WithStyles } from 'react-jss'
+import withStyles from 'react-jss'
+import { CartesianGrid, Line, LineChart, ResponsiveContainer, Tooltip, XAxis, YAxis } from 'recharts'
+import { DefaultTheme, type SaladTheme } from '../../../SaladTheme'
+import { formatBalance } from '../../../utils'
+import type { ChartDaysShowing, EarningPerMachine, EarningWindow } from '../../balance/models'
+import { earningsChartColors } from '../pages/constants'
+import { normalizeEarningsPerMachineData } from '../utils'
+
+const styles = (theme: SaladTheme) => ({
```
[the remark this answers is not in the document]
Comment (Maks19) [reply packages/web-app/src/modules/earn-views/components/EarningLineChart.tsx:11]: Done

File: packages/web-app/src/modules/earn-views/components/EarningChart.tsx (hunk 2 above)
Comment (Maks19) [reply packages/web-app/src/modules/earn-views/components/EarningChart.tsx:183]: Let's refactor EarningChart.tsx in another PR

**Review by vitto-moz — APPROVED** [review]

**Pull request merged**